File: backend/src/main/java/com/sliit/crave/service/FoodService.java
package com.sliit.crave.service;

import com.sliit.crave.dto.request.food.FoodRequest;
import com.sliit.crave.dto.response.Response;
import com.sliit.crave.dto.response.food.FoodResponse;

import java.util.List;

public interface FoodService {
    Response<List<FoodResponse>> getAllFoods();
    Response<FoodResponse> getFoodById(Long id);
    Response<FoodResponse> createFood(FoodRequest foodDTO);
    Response<FoodResponse> updateFood(Long id, FoodRequest foodDTO);
    Response<Void> deleteFood(Long id);
}

File: frontend/src/services/food.service.js
import {api} from "../api.js";

export const createFood = async (data) => {
    try {
        const response = await api.post("/foods", data);
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const getAllFoods = async () => {
    try {
        const response = await api.get("/foods");
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const getFoodById = async (foodId) => {
    try {
        const response = await api.get(`/foods/${foodId}`);
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const updateFood = async (foodId, data) => {
    try {
        const response = await api.put(`/foods/${foodId}`, data);
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const deleteFood = async (id) => {
    try {
        const response = await api.delete(`/foods/${id}`);
        return response.data;
    } catch (e) {
        console.error(e);
    }
};


